io.asgardio.java.oidc.sdk/src/main/java/io/asgardio/java/oidc/sdk/OIDCManagerImpl.java

```diff
@@ -0,0 +1,307 @@
+
+        AuthorizationErrorResponse errorResponse = authzResponse.toErrorResponse();
+        JSONObject responseObject = errorResponse.getErrorObject().toJSONObject();
+        logger.log(Level.INFO, "Error response object: ", responseObject);
+    }
+
+    private TokenResponse getTokenResponse(TokenRequest tokenRequest) {
+
+        TokenResponse tokenResponse = null;
+        try {
+            tokenResponse = TokenResponse.parse(tokenRequest.toHTTPRequest().send());
+        } catch (com.nimbusds.oauth2.sdk.ParseException | IOException e) {
+            logger.log(Level.ERROR, "Error while parsing token response.", e);
+        }
+        return tokenResponse;
+    }
+
+    private TokenRequest getTokenRequest(AuthorizationCode authorizationCode) {
+
+        URI callbackURI = oidcAgentConfig.getCallbackUrl();
+        AuthorizationGrant authorizationGrant = new AuthorizationCodeGrant(authorizationCode, callbackURI);
+        ClientID clientID = oidcAgentConfig.getConsumerKey();
+        Secret clientSecret = oidcAgentConfig.getConsumerSecret();
+        ClientAuthentication clientAuthentication = new ClientSecretBasic(clientID, clientSecret);
+        URI tokenEndpoint = oidcAgentConfig.getTokenEndpoint();
+
+        return new TokenRequest(tokenEndpoint, clientAuthentication, authorizationGrant);
+    }
+
+    private JSONObject requestToJson(AbstractRequest request) {
+
+        JSONObject obj = new JSONObject();
+        obj.appendField("tokenEndpoint", request.toHTTPRequest().getURI().toString());
+        obj.appendField("request body", request.toHTTPRequest().getQueryParameters());
+        return obj;
+    }
+
+    private Map<String, Object> getUserAttributes(String idToken) throws SSOAgentServerException {
+
+        Map<String, Object> userClaimValueMap = new HashMap<>();
+        try {
+            JWTClaimsSet claimsSet = SignedJWT.parse(idToken).getJWTClaimsSet();
+            Map<String, Object> customClaimValueMap = claimsSet.getClaims();
+
+            for (String claim : customClaimValueMap.keySet()) {
+                if (!SSOAgentConstants.OIDC_METADATA_CLAIMS.contains(claim)) {
+                    userClaimValueMap.put(claim, customClaimValueMap.get(claim));
+                }
+            }
+        } catch (ParseException e) {
+            throw new SSOAgentServerException(SSOAgentConstants.ErrorMessages.JWT_PARSE.getMessage(),
+                    SSOAgentConstants.ErrorMessages.JWT_PARSE.getCode(), e);
+        }
+        return userClaimValueMap;
+    }
+
+    private void validateConfig(OIDCAgentConfig oidcAgentConfig) throws SSOAgentClientException {
+
+        validateForCode(oidcAgentConfig);
+    }
+
+    private void validateForCode(OIDCAgentConfig oidcAgentConfig) throws SSOAgentClientException {
+
+        Scope scope = oidcAgentConfig.getScope();
+        if (scope.isEmpty() || !scope.contains(SSOAgentConstants.OIDC_OPENID)) {
+            logger.error("scope defined incorrectly.");
```

Comment:
Is it necessary to both throw and log?

Reply:
Resolved with https://github.com/asgardio/asgardio-java-oidc-sdk/pull/1/commits/87be1ad8b44e103c81d6ce8a6ed22d4f6fc4f854.